ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/workshopAdmin/presenter/WorkshopAdminPresenterImpl.java

```diff
@@ -0,0 +1,178 @@
+import org.iplantc.de.admin.desktop.client.workshopAdmin.gin.factory.WorkshopAdminViewFactory;
+import org.iplantc.de.admin.desktop.client.workshopAdmin.model.MemberProperties;
+import org.iplantc.de.admin.desktop.client.workshopAdmin.service.WorkshopAdminServiceFacade;
+import org.iplantc.de.admin.desktop.shared.Belphegor;
+import org.iplantc.de.client.models.collaborators.Collaborator;
+import org.iplantc.de.client.models.groups.GroupAutoBeanFactory;
+import org.iplantc.de.client.models.groups.Member;
+import org.iplantc.de.client.models.groups.MemberSaveResult;
+import org.iplantc.de.collaborators.client.events.UserSearchResultSelected;
+import org.iplantc.de.commons.client.ErrorHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author dennis
+ */
+public class WorkshopAdminPresenterImpl implements WorkshopAdminView.Presenter {
+
+    private final WorkshopAdminView view;
+    private final WorkshopAdminServiceFacade serviceFacade;
+    private final GroupAutoBeanFactory groupAutoBeanFactory;
+    private final WorkshopAdminView.WorkshopAdminViewAppearance appearance;
+    private final ListStore<Member> listStore;
+
+    private final class UserSearchResultSelectedEventHandler
+            implements UserSearchResultSelected.UserSearchResultSelectedEventHandler {
+
+        private Member memberFromCollaborator(Collaborator collaborator) {
+            Member member = groupAutoBeanFactory.getMember().as();
+            member.setId(collaborator.getUserName());
+            member.setAttributes(new ArrayList<String>());
+            member.setEmail(collaborator.getEmail());
+            member.setFirstName(collaborator.getFirstName());
+            member.setInstitution(collaborator.getInstitution());
+            member.setLastName(collaborator.getLastName());
+            member.setName(collaborator.getName());
+            return member;
+        }
+
+        private boolean listContainsMember(List<Member> members, Member newMember) {
+            String newMemberId = newMember.getId();
+            if (newMemberId != null) {
+                for (Member member : members) {
+                    if (member.getId() != null && member.getId().equals(newMemberId)) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        @Override
+        public void onUserSearchResultSelected(UserSearchResultSelected event) {
+
+            // Ignore the event if it wasn't initiated by the workshop admin panel.
+            if (!event.matchesTag(view.userSearchEventTag)) {
+                return;
+            }
+
+            // Add the user to the list if not there already.
+            Member member = memberFromCollaborator(event.getCollaborator());
+            if (!listContainsMember(listStore.getAll(), member)) {
+                listStore.add(member);
+            }
+        }
+    }
```

Comment:
I've seen other cases where the presenter implements all of the event handler interfaces instead. I considered using that approach here, but opted for inner classes instead just because inner classes give me more control over which methods can be called.